### PR TITLE
Aztec Passport

### PR DESCRIPTION
**Pull request by Envoy-VC**
# Aztec Passport

Aztec Passport is an identity verification platform that enables users to verify their social accounts, decentralized identifiers (DIDs), and on-chain activities while preserving privacy.

### How Aztec Passport leverages Aztec’s features:

- **Social Accounts Verification**: Uses zk-email proofs to validate ownership without exposing sensitive information.
- **DID Verification**: Utilizes zk-proofs to verify JWTs for secure validation of decentralized identifiers.
- **On-Chain Activity Verification**: Uses Aztec’s L1-to-L2 message portal to verify on-chain activities.

## Project Links

Entire Code for the project can be found [here](https://github.com/Envoy-VC/ab2/tree/aztec-passport/Aztec-Passport)

https://github.com/Envoy-VC/ab2/tree/aztec-passport/Aztec-Passport

as well as some of the Libraries created during building this Project as developer tooling:

- `noir_hmac`: HMAC implementation in Noir Language over SHA-256 Hash Algorithm.
- `noir_base64_lib`: Extension of `noir_base64` with support for Vectors and Base64 URL encoding and decoding
- `noir_jwt`: Verify JWTs and attest claims in Noir Language
- `noir_social_verify`: Wrapper around zkemailnr to verify social accounts and extract details like X username, and emails.

All of them can be found here

https://github.com/Envoy-VC/noir_hmac
https://github.com/Envoy-VC/noir_base64_lib
https://github.com/Envoy-VC/noir_jwt
https://github.com/Envoy-VC/noir_social_verify

## Video Demo
https://youtu.be/3IpaD-8wxCM

[Demo Video](https://www.youtube.com/watch?v=3IpaD-8wxCM)

### CODE REVIEW

**Comment by Envoy-VC** [comment]
# Aztec Passport

Aztec Passport is an identity verification platform that enables users to verify their social accounts, decentralized identifiers (DIDs), and on-chain activities while preserving privacy.

### How Aztec Passport leverages Aztec’s features:

- **Social Accounts Verification**: Uses zk-email proofs to validate ownership without exposing sensitive information.
- **DID Verification**: Utilizes zk-proofs to verify JWTs for secure validation of decentralized identifiers.
- **On-Chain Activity Verification**: Uses Aztec’s L1-to-L2 message portal to verify on-chain activities.

File: Aztec-Passport/README.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
+# Aztec Passport
+
+Aztec Passport is an identity verification platform that enables users to verify their social accounts, decentralized identifiers (DIDs), Biometrics and on-chain activities while preserving privacy.
+
+### How Aztec Passport leverages Aztec’s features:
+
+- **Social Accounts Verification**: Uses zk-email proofs to validate ownership without exposing sensitive information.
+- **DID Verification**: Utilizes zk-proofs to verify JWTs for secure validation of decentralized identifiers.
+- **Bio-metric Verification**: Use of face descriptors to verify government issued IDs such as Driver's License, Passport, etc.
+- **On-Chain Activity Verification**: Uses Aztec’s L1-to-L2 message portal to verify on-chain activities such as ETH Balance and ENS Names.
```
[the remark this answers is not in the document]
Comment (Envoy-VC) [reply Aztec-Passport/README.md:10]: All of the developer tooling and project has been created as part of Alpha Builder 2.

For now the application is agnostic to any use case. It provides a Humanity Score in range of `0-100` which is dynamic and changes as new services are added to protocol, on a high level each services has the following interface:

```noir
struct Service {
    address: AztecAddress,
    weight: u64,
    base_score: u64,
    max_score: u64,
}
```

and factors such as a `diminishing_factor`, `bonus_multiplier`, `no_of_verified_services` affect the Humanity score and appropriately Normalize it for any number of services.

So basically any application can use this score to power their use-case. For this Cohort only the passport service is built.

[the remark this answers is not in the document]
Comment (jzaki) [reply Aztec-Passport/README.md:10]: I noticed the later PR here -> https://github.com/AztecProtocol/ab2/pull/27. Are both needed?

[the remark this answers is not in the document]
Comment (Envoy-VC) [reply Aztec-Passport/README.md:10]: No both are the same, it was told we need a new PR for submission so it's basically same.

[the remark this answers is not in the document]
Comment (0xrafi) [reply Aztec-Passport/README.md:10]: sorry for the confusion there, you only need one PR. Feel free to update one to be your submission and close the other 🙏🏽

[the remark this answers is not in the document]
Comment (Envoy-VC) [reply Aztec-Passport/README.md:10]: > sorry for the confusion there, you only need one PR. Feel free to update one to be your submission and close the other 🙏🏽

Sure done 👍

File: .gitmodules [hunk 2]
```diff
@@ -0,0 +1,3 @@
+[submodule "Aztec-Passport/packages/l1-contracts/lib/forge-std"]
```
Comment (0xrafi) [inline .gitmodules:1]: could you make the gitmodules be nested inside of your project directory? all code/files for your project should be inside your named project folder, ty 🙏🏽

Comment (Envoy-VC) [reply .gitmodules]: @0xrafi done 👍
